llama_index/indices/base_retriever.py

```diff
@@ -28,3 +34,8 @@ def _retrieve(self, query_bundle: QueryBundle) -> List[NodeWithScore]:
 
         """
         pass
+
+    # TODO: make this abstract
+    # @abstractmethod
```

Comment:
any reason to leave this commented out?

Reply:
yeah. so that is a breaking change. Every `BaseRetriever` would have to implement it. Is that okay?

I can make all the changes and since we have mypy it will catch if we have parts that don't have this implemented.

Reply:
but having an async implementation for each that is actually async is a lot work, would take time

Reply:
Yea that makes sense. Really the only async part needed here is any embeddings or llm calls right?

Maybe by default, instead of returning an empty list, we should either call _retrieve() or raise a NotImplementedError? The latter might be too restrictive though

Reply:
we could but then we'll run into fake async issues - the API says the call is async but it's not and then you have to spend your time debugging only to realize that internally the call was blocking.

Reply:
Yea that's fair. Too many places to implement async (and sooo much code duplication, ew)